update databricks source setup page (#6342)
segmentio/segment-docs: src/connections/reverse-etl/reverse-etl-source-setup-guides/databricks-setup.md

Set up Databricks as your Reverse ETL source. 

At a high level, when you set up Databricks for Reverse ETL, the configured service-principal needs read permissions for any resources (databases, schemas, tables) the query needs to access. Segment keeps track of changes to your query results with a managed schema (`__SEGMENT_REVERSE_ETL`), which requires the configured service-principal to allow write permissions for that schema.

> info ""

> > Segment supports only OAuth (M2M) authentication. To generate client ID and Secret, follow the steps listed in the [Databricks docs: OAuth machine-to-machine (M2M) authentication](https://docs.databricks.com/en/dev-tools/auth/oauth-m2m.html){:target="_blank"}.

Reviewer: ```suggestion
> Segment supports only OAuth (M2M) authentication. To generate a client ID and Secret, follow the steps listed in Databricks' [OAuth machine-to-machine (M2M) authentication](https://docs.databricks.com/en/dev-tools/auth/oauth-m2m.html){:target="_blank"} documentation.
```

    * Hostname: `adb-xxxxxxx.azuredatabricks.net`
    * Http Path: `/sql/1.0/warehouses/xxxxxxxxx`
    * Port: `443` (default)
    * Service principal client ID: `<your client ID>`

>     * OAuth secret: `<OAuth secret to be used during connection>`

Reviewer: ```suggestion
    * OAuth secret: `<OAuth secret used during connection>`
```

>     * Catalog [optional]: `<if not specify, Segment will use the default catalog>`

Reviewer: ```suggestion
    * Catalog [optional]: If not specified, Segment will use the default `hive_metastore` catalog
```

11. Click **Test Connection** to see if the connection works. If the connection fails, make sure you have the right permissions and credentials, then try again.
12. Click **Add source** if the test connection is successful. 

> warning ""

> > Segment previously supported token-based authentication, but the authentication method has been changed based on Databricks' recommendation.

Reviewer: ```suggestion
> Segment previously supported token-based authentication, but now uses OAuth (M2M) authentication at the recommendation of Databricks.
```

> > If you set up your source in the past using a token, it will still be supported; however, for new source or for updating the connection settings of an existing source, we will only support [OAuth machine-to-machine (M2M) authentication](https://docs.databricks.com/en/dev-tools/auth/oauth-m2m.html){:target="_blank"}.

Reviewer: ```suggestion
> If you previously set up your source using token-based authentication, Segment will continue to support it. If you want to create a new source or update the connection settings of an existing source, Segment only supports [OAuth machine-to-machine (M2M) authentication](https://docs.databricks.com/en/dev-tools/auth/oauth-m2m.html){:target="_blank"}.
```
